### PR TITLE
insights: create migrations to clean up insights tables for 3.31 beta

### LINKED ISSUE #23785: insights: truncate tables with semantic changes for 3.31 release
The code insights tables
`series_points` in the codeinsights-db has had a semantic change to how points are stored, and should be cleared with the 3.31 release.
The `insights_query_runner_jobs` in the frontend db has historically been bloated full of useless records that needs to be cleared now that the code insights backend tracks state for series backfill.

Neither of these tables will result in material loss of data, these are by design and definition ephemeral and reproducible.

### PR DESCRIPTION
**Pull request by coury-clark**
Closes #23785

A little context:
Some of the tables that code insights use need to be reset as we go into 3.31 for beta. Some of this is due to the tables having a ton of garbage records blocking the queue, others are due to semantic changes in how we store the points, and others are to avoid any possible corruption caused by the 3.30 p1.

Adding distribution just to take a look and let me know if there is anything wrong with this plan for the 3.31 release. For example, is there any reason to think these migrations might block an upgrade if they were to fail?

### CODE REVIEW

**Comment by coury-clark** [comment]
`2021/08/11 16:39:29 MigrateDB: Failed to migrate the DB. Please contact support@sourcegraph.com for further assistance: ERROR: current transaction is aborted, commands ignored until end of transaction block (SQLSTATE 25P02) in line 0: SELECT pg_advisory_unlock($1)`

https://buildkite.com/sourcegraph/sourcegraph/builds/104536#7250b82e-baae-4dd5-a975-9d0c6b00fb76

Maybe this is a good answer to my question above 😬

File: migrations/codeinsights/1000000010_series_points_reset.up.sql [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+-- we are going to reset the stored data and all of the underlying Timescale chunks back to zero.
+-- Note: This data is by design reproducible, so there is no risk of permanent data loss here. Any and all data
+-- will be queued and regenerated as soon as code insights starts up.
+
+-- Drop all Timescale chunks prior to now. This will reduce a bloated number of partitions caused by old
+-- data generation patterns. This is a Timescale specific thing.
+SELECT drop_chunks('series_points', CURRENT_TIMESTAMP::DATE);
+
+-- Clean up the remaining records if any exist.
+TRUNCATE series_points;
```
Comment (efritz) [inline migrations/codeinsights/1000000010_series_points_reset.up.sql:22]: Try moving truncate out of the txn?

Comment (coury-clark) [reply migrations/codeinsights/1000000010_series_points_reset.up.sql]: Turns out in Postgres you have to manually specify `CASCADE` to get truncate to work with FK, and it also turns out when wrapped in the transaction there is no useful logging output!

**Comment by coury-clark** [comment]
Failing build is due to a percy flake

**Review by efritz — APPROVED** [review]

**Comment by efritz** [comment]
FYI, I've done similar things to the entirety of code intel data 😅

**Pull request merged**